src/runtime/vm/vm.cc

```diff
@@ -68,8 +68,17 @@ inline ObjectRef CopyTo(ObjectRef src, const DLContext& ctx) {
     if (nd_array->ctx.device_type != ctx.device_type) {
       return nd_array.CopyTo(ctx);
     }
+    return src;
+  } else {
+    CHECK(src->IsInstance<ADTObj>())
+        << "VM data must be NDArray or a list of NDArray, but received: " << src->_type_key;
+    std::vector<ObjectRef> ret;
+    ADT adt = Downcast<ADT>(src);
+    for (size_t i = 0; i < adt.size(); i++) {
+      ret.push_back(CopyTo(adt[i], ctx));
+    }
+    return ADT(0, ret.begin(), ret.end());
```

Comment:
why not use `adt->tag`?

Reply:
Yeah, for the input we only used Tuple whose tag is 0. But you are right, it's better to use tag.